src/main/java/com/powsybl/openloadflow/dc/DcTargetVector.java

```diff
@@ -24,7 +25,12 @@ public class DcTargetVector extends TargetVector<DcVariableType, DcEquationType>
     public static void init(Equation<DcVariableType, DcEquationType> equation, LfNetwork network, double[] targets) {
         switch (equation.getType()) {
             case BUS_TARGET_P:
-                targets[equation.getColumn()] = network.getBus(equation.getElementNum()).getTargetP();
+                LfBus bus = network.getBus(equation.getElementNum());
+                targets[equation.getColumn()] = bus.getTargetP();
```

Comment:
move `targets[equation.getColumn()] = bus.getTargetP();` in an `else` (=not slack) block below ?

Reply:
ah no this would be wrong, forget it.